pkg/cloud-provider/cloudapi/internal/accounts_common.go

```diff
@@ -181,7 +182,7 @@ func (accCfg *cloudAccountConfig) performInventorySync() error {
 			defer wg.Done()
 
 			hasFilters, isFilterNil := serviceCfg.hasFiltersConfigured()
-			if !hasFilters {
+			if !hasFilters && !serviceCfg.needPollWithoutFilter() {
```

Comment:
We need this check as needPollWithoutFilter() is true for ec2/computer service, however when we have other services(where needPollWithoutFilter() may need to be set to false), without any configured filters we wouldn't want to poll cloud inventory.

Reply:
In that case, we should have similar thing for azure. Currently we are hard coding it for azure.
func (computeCfg *computeServiceConfig) NeedPollWithoutFilter() bool {
	return true
}

Reply:
I want to enable inventory poll for computer service even when filters are not set, this is how i am achieving it, instead of hardcoding it as "true" in NeedPollWithoutFilter() , i am using pollVpc variable now.